testsuite/cluster-test/src/tx_emitter.rs

```diff
@@ -729,16 +739,18 @@ impl StatsAccumulator {
             submitted: self.submitted.load(Ordering::Relaxed),
             committed: self.committed.load(Ordering::Relaxed),
             expired: self.expired.load(Ordering::Relaxed),
+            latency: self.latency.load(Ordering::Relaxed),
         }
     }
 }
 
 impl TxStats {
-    pub fn rate(&self, window: Duration) -> TxStatsRate {
+    pub fn rate(&self, window: Duration, total_tx_num: u64) -> TxStatsRate {
```

Comment:
You don't need to supply `total_tx_num` as a parameter, I think you have all the information in the function

Reply:
My understanding for latency is per tx, so in this part, we pass in duration because we need tx number per sec, but for latency, I think we should calculate it by time.

for example, if we have 1000 txs in 10mins, we should divide total latency by 1000? So we can have a average latency for each transaction in last 10 mins?

Please correct me if I am misunderstanding.

Reply:
I c, I can use self directly instead of passing by value. Please ignore my last comment. Not very familiar with Rust ^_^ Sorry~~